### ISSUE TITLE
 Flexible raft: custom Quorum Size

### ISSUE THREAD

**Issue by killme2008** [sofastack/sofa-jraft#265]
https://github.com/eBay/NuRaft/blob/master/docs/custom_quorum_size.md

**Comment by steven05jiang** [comment]
Just wondering do you have a plan for this feature? 

In my use case, I'm looking for a way to make sure the quorum size is 3 for a 5-peer deployment when there are only 3 or fewer peers in the configuration. There could be some reasons why not all peers in the configurations. For example, the peer is invisible if it can't be deployed as the limit on the host resources while its IP is not static.

**Comment by killme2008** [comment]
@steven05jiang  Yep, we have a plan, but it's not urgent right now, we don't put it into 1.3.0 milestone, maybe we will implement it in next release.Please keep attention on this issue.

**Comment by steven05jiang** [comment]
@killme2008 Thx for the update! Looking forward to it.

**Comment by shihuili1218** [comment]
这个思路听起来是可以行的，并且它可以成为jraft的一大特性。但是目前好像没有看到这一设计应用于生产，jraft要应用这一特性，我们可能要制定详细的测试方案，这一块我没有好的思路。
另外，jraft是支持成员变更的，写死R=4,W=2应该不行的，我的想法是配置使用一个小数来配置，例如R=0.8，W=0.4

**Comment by shihuili1218** [comment]
或者可以考虑作为一个测试分支，默认关闭，在小范围使用中印证是否正确
